tests/test_meteodatalab/test_ogd_api.py

```diff
@@ -111,23 +111,48 @@ def test_download_from_ogd(
     mock_session: mock.MagicMock, tmp_path: Path, with_headers: bool
 ):
     content = b"some content"
-    href = "https://test.com/path/to/some-file.grib"
-    body = {
-        "features": [{"assets": {"some-asset.ext": {"href": href}}}],
+    main_href = "https://test.com/path/to/some-file.grib"
+    horizontal_href = "https://test.com/path/to/horizontal.grib"
+    vertical_href = "https://test.com/path/to/vertical.grib"
+
+    mock_post_response = mock.Mock()
+    mock_post_response.json.return_value = {
+        "features": [{"assets": {"some-asset.ext": {"href": main_href}}}],
         "links": [],
     }
-    mock_post_response = mock.Mock(**{"json.return_value": body})
+    mock_post_response.raise_for_status = mock.Mock()
+
     if with_headers:
         headers = {"X-Amz-Meta-Sha256": hashlib.sha256(content).hexdigest()}
     else:
         headers = {}
-    mock_get_response = mock.Mock(
-        headers=headers, **{"iter_content.return_value": [content]}
-    )
+
+    def mock_get_response(url, *args, **kwargs):
+        # Respond with asset list for coordinate URLs
+        if "assets" in url:
+            mock_resp = mock.Mock()
+            mock_resp.raise_for_status = mock.Mock()
+            mock_resp.json.return_value = {
+                "assets": {
+                    "horizontal_constants_icon-ch2-eps.grib2": {
+                        "href": horizontal_href
+                    },
+                    "vertical_constants_icon-ch2-eps.grib2": {"href": vertical_href},
+                }
+            }
+            return mock_resp
+
+        # Return mock file download
+        mock_resp = mock.Mock()
+        mock_resp.raise_for_status = mock.Mock()
+        mock_resp.iter_content.return_value = [content]
```

Comment:
It could be good to have different content for different flies to make sure that they don't get swapped.